### PR TITLE
Badge modals on user profile

### PR DESCRIPTION
**Pull request by katiecrane**
### What does this PR do?

This PR completes work for badges in the user profile. A user is able to click on a badge to get more information on how they earned it, or information on how to earn it if they haven't yet.

This involves a decent amount of refactoring on the `<Badge>` component:
- All the text displayed with a badge is now passed as children. 
- `<BaseFigure>` is now used instead of `<Figure>`

#### Small
image

#### Medium
image

#### Large
image

#### Opening and closing modal
modal_profile

### Any background context you want to provide?

Copy and wireframes are linked in the card!

### What are the relevant tickets/cards?

Refs [Pivotal ID #165719236](https://www.pivotaltracker.com/story/show/165719236)

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [x] Added screenshot to PR description of related front-end updates on **small** screens.
* [x] Added screenshot to PR description of related front-end updates on **medium** screens.
* [x] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

File: resources/assets/components/Affirmation/Affirmation.js [hunk 1]
```diff
@@ -54,15 +54,19 @@ const Affirmation = ({
...
                   earned
-                  boldText
                   className="badge padded"
                   size="medium"
                   name="signupBadge"
-                  text="1 Sign-Up"
-                  explainerText="Congratulations! You signed up for your first campaign...and
+                >
+                  <p>
+                    <strong>1 Sign-Up</strong>
```
Comment (DFurnes) [inline resources/assets/components/Affirmation/Affirmation.js]: Would it make sense to turn this into a `h4`? (It should have pretty much the same appearance as bolded paragraph text, but better semantic meaning for screen readers and the like).

File: resources/assets/components/pages/AccountPage/Badge.js [hunk 2]
```diff
@@ -1,49 +1,48 @@
...
-    earned,
-    boldText,
-    explainerText,
-    size,
-    className,
-  } = props;
+  const { name, earned, size, className, children, showLock } = props;
   const badgeImageIndex = earned ? name : `${name}Locked`;
 
+  const media = showLock ? (
```
Comment (DFurnes) [inline resources/assets/components/pages/AccountPage/Badge.js]: This is kinda subjective, but it's probably more readable to move the `showLock` ternary around to the lock image itself, and keep the `position: relative` wrapper in both cases (since we DRY up the otherwise duplicated badge image).

Comment (DFurnes) [inline resources/assets/components/pages/AccountPage/Badge.js]: Another subjective suggestion – it might make more sense to have the `showLock` parameter determine whether the lock should be displayed for unearned badges (so `showLock && !earned` here). Then it can just be a constant and there's only the one `earned` prop that's being flipped.

File: resources/assets/components/pages/AccountPage/BadgesTab.js [hunk 3]
```diff
@@ -3,165 +3,380 @@ import PropTypes from 'prop-types';
...
 
-const NEWSLETTER_BADGE = gql`
-  query SubscriptionTopicsBadgeQuery($userId: String!) {
-    user(id: $userId) {
-      emailSubscriptionTopics
-    }
+  showModal(name, earned, title, text) {
+    this.setState({
+      showModal: true,
+      modalName: name,
```
Comment (DFurnes) [inline resources/assets/components/pages/AccountPage/BadgesTab.js]: Instead of providing these four different state variables in order to toggle the modal, it'd be nice to have an object we could reference (so, for example, we might just set `{ showModal: true, modalName: 'signupBadge' }` and we'd reference the title, text, and "earned" criteria from a centralized place.

(And actually, we could even simplify that to `{ modal: 'signupBadge' } or ` { modal: null }` then to toggle!)

File: resources/assets/components/pages/AccountPage/BadgesTab.js [hunk 4]
```diff
@@ -3,165 +3,380 @@ import PropTypes from 'prop-types';
...
-                  name="voterBadge"
-                  text="Registered Voter"
-                />
-              </li>
-            )}
-          </Query>
-        </ul>
+
+  render() {
+    const SIGNUP_COUNT_BADGE = gql`
```
Comment (DFurnes) [inline resources/assets/components/pages/AccountPage/BadgesTab.js]: Let's pull these queries out of the render function since they're constants.

File: resources/assets/components/Affirmation/Affirmation.js (hunk 1 above)
Comment (katiecrane) [reply resources/assets/components/Affirmation/Affirmation.js]: Can confirm it looks _exactly_ the same!

File: resources/assets/components/pages/AccountPage/Badge.js (hunk 2 above)
Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/Badge.js]: @DFurnes I'm a bit confused about what you mean - we don't want to display the lock for all `unearned` badges, that's why it's a separate parameter right now so I don't think that can be collapsed into one value?

File: resources/assets/components/pages/AccountPage/BadgesTab.js (hunk 3 above)
Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/BadgesTab.js]: So we would have several objects like this (one for each badge):
```
const signupBadge = {
  title: '1 SIGN-UP',
  earnedText:
    'Hey, congrats! You earned a badge for signing up for your first campaign. More badges ahead…!',
  unearnedText: 'Unlock this badge by signing up for a campaign.',
};
```
And then state would just be:
```
 this.state = {
      modalName: '',
      modalEarned: false,
    };
```
And if `this.state.modalName` then we could pull in the correct fields from the corresponding object?

File: resources/assets/components/pages/AccountPage/Badge.js (hunk 2 above)
Comment (DFurnes) [reply resources/assets/components/pages/AccountPage/Badge.js]: Sorry that wasn't clear at all! 🙃Let me try again!

I was thinking it might be easier to set `showLock={true}` for any badge where it _would_ show a lock if unearned (and if so, then calculate whether or not to show the lock _inside_ the Badge), rather than `earned={earned} showLock={!earned}`.

File: resources/assets/components/pages/AccountPage/BadgesTab.js (hunk 3 above)
Comment (DFurnes) [reply resources/assets/components/pages/AccountPage/BadgesTab.js]: I was thinking like a big object!!

```js
const badges = {
  signupBadge: {
    title: '1 SIGN-UP',
    earnedText: 'Hey, congrats! .....',
    unearnedText: 'Unlock this badge by…',
  },
  onePostBadge: {
    title: '1 ACTION',
    // and so on...
};
```

Then, we could grab the appropriate title/text like `badges[this.state.modalName].title`!

Comment (DFurnes) [reply resources/assets/components/pages/AccountPage/BadgesTab.js]: And actually, if we included the `query` and `earned` callback in the object, we might even be able to build the grid of icons as a loop over the object?!

```js
const badges = {
  signupBadge: {
    title: '1 SIGN-UP',
    earnedText: 'Hey, congrats! .....',
    unearnedText: 'Unlock this badge by…',
    query: SIGNUP_COUNT_BADGE,
    earned: data => data.signupsCount > 0,
  },
  onePostBadge: {
    title: '1 ACTION',
    // and so on...
};
```

Definitely not a merge blocker, but a cool way to DRY this interface up if it works out.

File: resources/assets/components/pages/AccountPage/Badge.js (hunk 2 above)
Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/Badge.js]: OH okay I follow now! It would simplify what we're getting when we create the badge in `<BadgeModal>`. Thanks!

File: resources/assets/components/pages/AccountPage/BadgesTab.js (hunk 3 above)
Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/BadgesTab.js]: AH one big object awesome! I'll go with that part and save the looping over the object for another time

File: resources/assets/components/Affirmation/Affirmation.js (hunk 1 above)
Comment (katiecrane) [reply resources/assets/components/Affirmation/Affirmation.js]: Fixed in https://github.com/DoSomething/phoenix-next/pull/1525/commits/bd38600ce37fe1a368d35520512bfd0f3f7b3b94

File: resources/assets/components/pages/AccountPage/Badge.js (hunk 2 above)
Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/Badge.js]: Fixed in https://github.com/DoSomething/phoenix-next/pull/1525/commits/bbf89383be6568bc2f2330aaff7f427d4124960c

File: resources/assets/components/pages/AccountPage/BadgesTab.js (hunk 4 above)
Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/BadgesTab.js]: Fixed in https://github.com/DoSomething/phoenix-next/pull/1525/commits/1d5bb57510eed3c0a8c70adb3920562b31d72405

File: resources/assets/components/pages/AccountPage/Badge.js (hunk 2 above)
Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/Badge.js]: Fixed in https://github.com/DoSomething/phoenix-next/pull/1525/commits/0525ff6d6ddd0677b8b64dd4ffab867a5f443327

File: resources/assets/components/pages/AccountPage/BadgesTab.js (hunk 3 above)
Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/BadgesTab.js]: Fixed in https://github.com/DoSomething/phoenix-next/pull/1525/commits/c14165b78b4fa8bc5fe01d38c09bb65c6e7801bd

**Comment by katiecrane** [comment]
@DFurnes suggestions have all been addressed, some bigger refactoring so I think this could benefit from 👀 again, thank you!

**Comment by katiecrane** [comment]
Tested on review app woohoo here we go!
